contracts/AllDay.cdc

```diff
@@ -558,6 +569,269 @@ access(all) contract AllDay: NonFungibleToken {
         AllDay.account.storage.save(setPlayTierMap, to: /storage/AllDayAdminSetPlayTierMap)
     }
 
+    //------------------------------------------------------------
+    // Badges
+    //------------------------------------------------------------  
+
+    access(all) struct Badge{
+        access(all) var slug: String
+        access(all) var title: String
+        access(all) var description: String
+        access(all) var visible: Bool
+        access(all) var slugV2: String
+        access(all) var metadata: {String: String}
+
+        init(slug: String, title: String, description: String, visible: Bool, slugV2: String){
```

Comment:
Nit: can be a view function (not modifying any state), same for init in Badges resource